packages/phone-number-privacy/signer/src/pnp/services/quota.ts

```diff
@@ -65,52 +64,84 @@ export abstract class PnpQuotaService
       ['getQuotaStatus', session.request.url]
     )
 
-    let hadBlockchainError = false
-    let performedQueryCount = -1
-    let totalQuota = -1
-    let blockNumber = -1
+    const quotaStatus: PnpQuotaStatus = {
+      totalQuota: -1,
+      performedQueryCount: -1,
+      blockNumber: undefined,
+    }
     if (performedQueryCountResult.status === 'fulfilled') {
-      performedQueryCount = performedQueryCountResult.value
+      quotaStatus.performedQueryCount = performedQueryCountResult.value
     } else {
-      session.logger.error(performedQueryCountResult.reason)
-      session.errors.push(ErrorMessage.DATABASE_GET_FAILURE)
+      session.logger.error(
+        { error: performedQueryCountResult.reason },
+        'failed to get performedQueryCount from db'
```

Comment:
The only reason would be to avoid logging the ErrorMessage too many times so that our alerting thresholds are easier to reason about. If we log every db failure multiple times, then it's harder to know how many db failures should trigger an on-call alert (all the session.errors get logged when we send the response). That being said, we're not super consistent about this and I think it does make sense to use the error message here. Having some duplicate logging is okay, as long as we're mindful about it so that it doesn't get out of hand (as it has in the past - causing us to think one db failure might actually equal dozens of failures)